FWCore/Framework/src/EventSetupRecordProvider.cc

```diff
@@ -133,97 +119,121 @@ namespace edm {
       typedef DataProxyProvider::KeyedProxies ProxyList;
       typedef EventSetupRecordProvider::DataToPreferredProviderMap PreferredMap;
 
-      const ProxyList& keyedProxies(iProvider->keyedProxies(this->key()));
-      ProxyList::const_iterator finishedProxyList(keyedProxies.end());
-      for (ProxyList::const_iterator keyedProxy(keyedProxies.begin()); keyedProxy != finishedProxyList; ++keyedProxy) {
-        PreferredMap::const_iterator itFound = iMap.find(keyedProxy->first);
-        if (iMap.end() != itFound) {
-          if (itFound->second.type_ != keyedProxy->second->providerDescription()->type_ ||
-              itFound->second.label_ != keyedProxy->second->providerDescription()->label_) {
-            //this is not the preferred provider
-            continue;
+      for (unsigned int iovIndex = 0; iovIndex < nConcurrentIOVs_; ++iovIndex) {
```

Comment:
Change to `for(auto& record : recordImpls_) {`
and then get the iovIndex from the record. Then at line 135 you only have to do
`record.add(keyedProxy.dataKey(), keyedProxy.dataProxy());`

Reply:
Done.